extensions/mqtt/processors/PublishMQTT.h

```diff
@@ -74,19 +77,21 @@ class PublishMQTT : public processors::AbstractMQTTProcessor {
       status_ = 0;
       read_size_ = 0;
     }
-    ~ReadCallback() = default;
-    int64_t process(const std::shared_ptr<io::BaseStream>& stream) {
+    ~ReadCallback() override = default;
+    int64_t process(const std::shared_ptr<io::BaseStream>& stream) override {
       if (flow_size_ < max_seg_size_)
         max_seg_size_ = flow_size_;
+      gsl_Expects(max_seg_size_ < gsl::narrow<uint64_t>(std::numeric_limits<int>::max()));
       std::vector<unsigned char> buffer;
       buffer.reserve(max_seg_size_);
       read_size_ = 0;
       status_ = 0;
       while (read_size_ < flow_size_) {
-        int readRet = stream->read(&buffer[0], max_seg_size_);
+        // MQTTClient_message::payloadlen is int, so we can't handle 2GB+
+        int readRet = static_cast<int>(stream->read(&buffer[0], max_seg_size_));
```

Comment:
we could check the return value of `stream->read(...)` with `io::isError` before casting to `int`

Reply:
done in 7ef4b05